Adds support for native OSX setup

maxieduncan opened the pull request:
Splits out common functionality into separate scripts that are shared by vagrant and native installations
The git repositories can now be checked out into the eclipse workspace, this allows them to be kept workspace specific and multiple workspaces to be generated
Moves the Team Project Set into the workspace to make it easier to access via eclipse

Adds a more detailed README

Adds support for native OSX installation. Requires PostgreSQL and Eclipse to already be installed, setup.sh script then:
* checks out SolarNetwork code
* configures PostgreSQL DB
* sets up eclipse workspace

Vagrant Improvements:
* Auto configures the VB tools for auto scaling the screen resolution

Future improvements to consider:
* Add dras SQL to vagrant setup
* Native support for Windows and/or Linux
* Make SolarNet projects configurable
* Make branch being checked out configurable

maxieduncan committed:
Adds support for native OSX setup

Splits out common functionality into separate scripts that are shared by vagrant and native installations
The git repositories can now be checked out into the eclipse workspace, this allows them to be kept workspace specific and multiple workspaces to be generated
Moves the Team Project Set into the workspace to make it easier to access via eclipse

Adds a more detailed README

Adds support for native OSX installation. Requires PostgreSQL and Eclipse to already be installed, setup.sh script then:
* checks out SolarNetwork code
* configures PostgreSQL DB
* sets up eclipse workspace

Vagrant Improvements:
* Auto configures the VB tools for auto scaling the screen resolution

Future improvements to consider:
* Add dras SQL to vagrant setup
* Native support for Windows and/or Linux
* Make SolarNet projects configurable
* Make branch being checked out configurable

msqr commented on osx/setup-db.sh:
```diff
@@ -0,0 +1,54 @@
+psql -U postgres -d postgres -c "alter user solarnet_test with password 'solarnet_test';"
+createdb -E UNICODE -l C -T template0 -O solarnet_test solarnet_unittest
+createlang plv8 solarnet_unittest
+psql -U postgres -d solarnet_unittest -c "CREATE EXTENSION IF NOT EXISTS citext WITH SCHEMA public;"
```
Also here's where you need to add another `CREATE EXTENSION` call for the `pgcrypto` extension.

msqr commented on osx/setup-db.sh:
```diff
@@ -0,0 +1,54 @@
+psql -U postgres -d postgres -c "alter user solarnet with password 'solarnet';"
+createdb -E UNICODE -l C -T template0 -O solarnet solarnetwork
+createlang plv8 solarnetwork
+psql -U postgres -d solarnetwork -c "CREATE EXTENSION IF NOT EXISTS citext WITH SCHEMA public;"
```
Here's where you need to add another `CREATE EXTENSION` call for the `pgcrypto` extension.

msqr commented on osx/setup-db.sh:
```diff
@@ -0,0 +1,54 @@
+# for some reason, plv8 often chokes on the inline comments, so strip them out
+sed -e '/^\/\*/d' -e '/^ \*/d' postgres-init-plv8.sql | psql -d solarnetwork -U postgres
+psql -d solarnetwork -U solarnet -f postgres-init.sql
+psql -d solarnetwork -U solarnet -f postgres-init-data.sql
```
Looks like you added the loading of `postgres-init-data.sql` but that wasn't part of the original script. The DDL no longer works, and would require updates to get it working. I don't think it is necessary to load.

msqr reviewed (commented):
Just adding some comments re discussion on Skype with postgres init not working.

maxieduncan committed:
Adds pgcrypto extension to PostgreSQL setup
Drops postgres-init-data.sql for loading initial data as it's no longer supported

maxieduncan committed:
Updated SolarNetwork.launch to include additional required osgi bundles
Moved "osx" scripts to bin as they're actually fairly generic

maxieduncan committed:
Removing the "solarnetwork-node/net.solarnetwork.node.system.ssh" from the workspace by default as it causes a number of errors to be logged at runtime
